src/commandHandlers/reminder.ts

```diff
@@ -0,0 +1,62 @@
+import { MessageEmbed, Message } from 'discord.js';
+import { scheduleJob } from '../scheduler.service';
+import { log } from '../logger';
+
+const ARG_SEPARATOR = '||';
+const CLOCK_EMOJI = ':alarm_clock:';
+
+/**
+ * This command lets the user set reminders for themselves, specifying the message they want to receive and the date
+ * and time they want to receive it.
+ */
+export const command = async (arg: string, embed: MessageEmbed, message: Message) => {
+    const [reminderMessage, dateString] = arg.split(ARG_SEPARATOR);
+
+    if (!reminderMessage || !dateString) {
+        return buildErrorMessage('Missing arguments');
+    }
+
+    try {
+        // TODO: maybe use a 3rd party package (e.g. moment) to parse more date formats
+        const dateTimestamp = Date.parse(dateString);
```

Comment:
To add to the `TODO` part, in the future another feature could be to say ` ... || 1 hour` so the reminder will be 1 hour from now

Reply:
Actually to make it easier and better for the user, we should just use `1 day` or `1 hour` from now, then we don't have to worry about the user's timezone, what do you think?

Reply:
I agree this "date and time" can be improved, it's a better UX to say "1 day" 👍. That is why I left the TODO, I'm not sure how to improve this parsing with those features you mentioned, at least right now 😅